packages/@react-spectrum/overlays/src/Popover.tsx

```diff
@@ -102,6 +102,49 @@ const PopoverWrapper = forwardRef((props: PopoverWrapperProps, ref: RefObject<HT
   let {modalProps} = useModal({
     isDisabled: isNonModal
   });
+  let [isHideArrow, setIsHideArrow] = useState(hideArrow);
+
+  useLayoutEffect(() => {
+    // check position of arrow and popover proximety to boundry edge
+    if (ref.current && arrowProps) {
+      let propsHideArrow = hideArrow;
+
+      // figure out the location of the offset and which popover dimension to use
```

Comment:
```suggestion
      // figure out the main (or is it the coss?) axis of the popover
```
which popover dimension to use for what?